### PR TITLE
refactor: optimize CSE main script for go template

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

- Remove unnecessary bash `if` blocks (including a few stragglers in `cse_config.sh`
- Remove unnecessary `AZURE_POLICY_ADDON` bash var via ARM (we can get this value via the ContainerService context of the go template layer directly at template generation time)
- ditto `TARGET_ENVIRONMENT`
- ditto `PRIVATE_AZURE_REGISTRY_SERVER`
- add consts for re-use of static file paths, and refer to them in the whole codebase, including go template, where appropriate
- optimized all dualstack foo under the `ensureDHCPv6` func

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_main.sh [hunk 1]
```diff
@@ -224,23 +201,23 @@ if $FULL_INSTALL_REQUIRED; then
...
 fi
 {{end}}
 
 if $REBOOTREQUIRED; then
-  echo 'reboot required, rebooting node in 1 minute'
-  /bin/bash -c "shutdown -r 1 &"
-  if [[ $OS == $UBUNTU_OS_NAME ]]; then
-      aptmarkWALinuxAgent unhold &
-  fi
+    echo 'reboot required, rebooting node in 1 minute'
```
Comment (mboersma) [inline parts/k8s/cloud-init/artifacts/cse_main.sh]: I prefer 2-space indentation in shell scripts (and I think that's more common in our codebase), but ideally we would agree on a standard. (See #2297.)

Comment (jackfrancis) [reply parts/k8s/cloud-init/artifacts/cse_main.sh]: My IDE was detecting that the large majority of indentation was 4 whitespace, so I applied that standard uniformly.

Happy to schedule a follow-up thread to discuss an abstract standard, and then we can apply that standard to shell scripts generally.

File: pkg/engine/armvariables_test.go [hunk 2]
```diff
@@ -116,8 +115,7 @@ func TestK8sVars(t *testing.T) {
 		"' KMS_PROVIDER_VAULT_NAME=',variables('clusterKeyVaultName')," +
 		"' IS_HOSTED_MASTER=%t" +
 		" IS_IPV6_DUALSTACK_FEATURE_ENABLED=%t" +
-		" PRIVATE_AZURE_REGISTRY_SERVER=',parameters('privateAzureRegistryServer')," +
-		"' AUTHENTICATION_METHOD=',variables('customCloudAuthenticationMethod')," +
+		" AUTHENTICATION_METHOD=',variables('customCloudAuthenticationMethod')," +
```
[the remark this answers is not in the document]
Comment (jackfrancis) [reply pkg/engine/armvariables_test.go]: Right. It's sort of unfortunate. The `'` that we see as the first character in most of the newline-arranged strings actually terminates an ARM string expression from the preceding line. In the case above, the omission of `PRIVATE_AZURE_REGISTRY_SERVER` means that `AUTHENTICATION_METHOD` now immediatly follows `IS_IPV6_DUALSTACK_FEATURE_ENABLED`, which is in fact one of the exceptions: it is *not* composed via an ARM expression; thus the `'` is not needed.

Yet more reasons why eliminating these unnecessary env vars altogether will be beneficial.

[the remark this answers is not in the document]
Comment (mboersma) [reply pkg/engine/armvariables_test.go]: Got it. Makes sense, but obviously a bit hard to read (which always happens with ARM expressions in code.)

**Pull request merged**